### ISSUE TITLE
Add support for Bzlmod

### ISSUE THREAD

**Issue by Vertexwahn** [Vertexwahn/rules_qt6#17]
Make `rules_qt6` great again! Add support for Bzlmod!

**Comment by Wito-1** [comment]
Hello @Vertexwahn, this is a great repository.

I've taken this repo made some modifications to do a few things:
1. upgrade to bzlmod 
2. download the qt libs using `aqt` 
3. modified the QtToolchainInfo / gen_* rules to be able to run in a sandbox environment

Are you accepting contributions?

**Comment by Vertexwahn** [comment]
@Wito-1 
Yes, I definitely accept contributions!
What you describe sounds perfect!
For me, it is also not so important to keep the old traditional WORKSPACE approach - if this does not work anymore - no problem - we can delete it. Maybe I can do a last release that people can reference for using those rules without Bzlmod, before we delete this approach.

**Comment by Sayter99** [comment]
Hi @Vertexwahn, I came across this issue and started trying to migrate rules_qt6 to use bzlmod ([my attempt](https://github.com/Sayter99/rules_qt6)). However, I found a weird thing that the compilation time increased a lot after migration.

Here's a [test repository](https://github.com/Sayter99/qwt_bazel). By building `//qwt` with bzlmod, it takes ~3 mins, but only ~1 min using WORKSPACE approach.

Noted that it only happens on Windows, (linux behaves the same), so I guess long paths might play a role here, but have no idea how to confirm this assumption. Any thoughts/ideas?

**Comment by Vertexwahn** [comment]
@Sayter99 I have no clue. Maybe you repost this in the bzlmod channel of Bazel on Slack

**Comment by Sayter99** [comment]
good idea, I've posted one https://bazelbuild.slack.com/archives/C014RARENH0/p1717470840590809, FYI.

**Comment by Vertexwahn** [comment]
Bzlmod support is now in place thanks to @Sayter99!

@Wito-1 Using aqt to download Qt is a great idea an would solve several problems (such as https://github.com/Vertexwahn/rules_qt6/issues/18) - will create a new ticket for this

**Comment by Sayter99** [comment]
Cool!

@Vertexwahn I was expecting you are going to build Qt w/o webengine support on M1, that could reduce the size a lot, but up to you.

`aqt` looks promising!!!
Just to share some of my thoughts on it, Ideally, we can let users to register the toolchain by providing version as well as components. I found the pattern in this [repo](https://github.com/hexdae/toolchains_arm_gnu.git) is worth looking into. For each version/platform, it provides required metadata so that the bazel rules will be generated, for example, [to fetch](https://github.com/hexdae/toolchains_arm_gnu/blob/master/toolchain/archives/arm_none_eabi.bzl), [rules generation](https://github.com/hexdae/toolchains_arm_gnu/tree/master/toolchain/templates).

With more information passed to templates as the approach by toolchains_arm_gnu, such as version number, it is possible to enable more features easily in the future, for example, private headers support (making version (6.8.0) to a variable) below
```python
[
    cc_library(
        name = "qt_%s_linux" % name,
        srcs = glob([
            "lib/lib%s.so*" % library_name,
            "lib/libicu*.so*",
        ]),
        hdrs = glob([
            "include/%s/**" % include_folder,
        ]),
        includes = [
            "include",
            "include/%s" % include_folder,
            "include/%s/6.8.0" % include_folder,
            "include/%s/6.8.0/%s" % (include_folder, include_folder),
            "include/%s/6.8.0/%s/private" % (include_folder, include_folder),
        ],
        target_compatible_with = ["@platforms//os:linux"],
        visibility = ["//visibility:public"],
    )
    for name, include_folder, library_name, _ in QT_LIBRARIES
]
```
